queries.sql

```diff
@@ -1,34 +1,88 @@
 /*Queries that provide answers to the questions from all projects.*/
 
-/* To find all animals whose name ends in "mon" */
+/* Update the species column to all animals to unspecified */
+BEGIN;
+UPDATE animals SET species = 'unspecified';
 
-SELECT * FROM animals WHERE name LIKE '%mon';
+/* Verify that the change was made */
+SELECT * FROM animals;
 
-/* To list the name of all animals born between 2016 and 2019 */
+/* Rollback the transaction */
+ROLLBACK;
 
-SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
+/* Verify that the "species" column went back to the state before the transaction */
+SELECT * FROM animals;
 
-/* To list the name of all animals that are neutered and have less than 3 escape attempts */
+/* Inside a transaction, update the "species" column of all animals with a name ending in "mon" to "digimon" */
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
 
-SELECT name FROM animals WHERE neutered = true AND escape_attempts < 3;
+/* Update the "species" column of all animals that don't have a species already set to "pokemon" */
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
 
-/* To list the date of birth of all animals named either "Agumon" or "Pikachu" */
+/* Commit the transaction */
+COMMIT;
 
-SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
+/* Verify that the change was made and persists after commit */
+SELECT * FROM animals;
 
-/* To list name and escape attempts of animals that weigh more than 10.5kg */
+/* Inside a transaction, delete all records in the "animals" table */
+BEGIN;
+DELETE FROM animals;
 
-SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+/* Roll back the transaction */
+ROLLBACK;
 
-/* To find all animals that are neutered */
+/* Verify that all records in the "animals" table still exist */
+SELECT * FROM animals;
 
-SELECT * FROM animals WHERE neutered = true;
+/* Delete all animals born after Jan 1st, 2022 */
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
 
-/* To find all animals not named Gabumon */
+/* Create a savepoint for the transaction */
+SAVEPOINT DELETE_ANIMALS_DOB;
 
-SELECT * FROM animals WHERE name != 'Gabumon';
+/* Update all animals' weight to be their weight multiplied by -1 */
+UPDATE animals SET weight_kg = -1 * weight_kg;
 
-/* To find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg) */
+/* Rollback to the savepoint */
+ROLLBACK TO DELETE_ANIMALS_DOB;
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+/* Update all animals' weights that are negative to be their weight multiplied by -1 */
+UPDATE animals SET weight = -1 * weight WHERE weight < 0;
 
+/* Commit the transaction */
+COMMIT;
+
+/* Write queries to answer the following questions */
+
+BEGIN;
```

Comment:
- Kindly I suggest you remove this `BEGIN` as there is no transaction for it :+1:

Reply:
Hi @Olamarx I have removed the BEGIN statement